bluetooth_mesh/messages/silvair/rrule_scheduler.py

```diff
@@ -0,0 +1,241 @@
+#
+# python-bluetooth-mesh - Bluetooth Mesh for Python
+#
+# Copyright (C) 2022  SILVAIR sp. z o.o.
+#
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+#
+from enum import IntEnum
+from bluetooth_mesh.messages.util import EnumAdapter
+from bluetooth_mesh.messages.util import EnumSwitch as Switch
+from bluetooth_mesh.messages.util import Opcode, SwitchStruct
+
+from construct import (
+    BitsInteger,
+    ByteSwapped,
+    BitStruct,
+    GreedyRange,
+    Int8sl,
+    Int8ul,
+    Int16sl,
+    Int16ul,
+    Int32ul,
+    PrefixedArray,
+    Struct,
+    Optional,
+    this
+)
+
+
+class RRuleSchedulerOpcode(IntEnum):
+    SILVAIR_RRULE_SCHEDULER = 0xE83601
+
+
+class RRuleSchedulerSubOpcode(IntEnum):
+    RULES_LIST_GET = 0x00
+    RULES_LIST_STATUS = 0x01
+    REGISTER_MAX_SIZE_GET = 0x02
+    REGISTER_MAX_SIZE_STATUS = 0x03
+    SCHEDULER_MODE_GET = 0x04
+    SCHEDULER_MODE_SET = 0x05
+    SCHEDULER_MODE_STATUS = 0x06
+    SCHEDULE_REGISTER_ENTRY_GET = 0x07
+    SCHEDULE_REGISTER_ENTRY_SET = 0x08
+    SCHEDULE_REGISTER_ENTRY_DELETE = 0x09
+    SCHEDULE_REGISTER_ENTRY_DELETE_ALL = 0x0A
+    SCHEDULE_REGISTER_ENTRY_STATUS = 0x0B
+    SCHEDULE_REGISTER_LIST_GET = 0x0C
+    SCHEDULE_REGISTER_LIST_STATUS = 0x0D
+
+
+class SchedulerMode(IntEnum):
+    DISABLED = 0x00
+    ENABLED = 0x01
+
+
+class Day(IntEnum):
```

Comment:
Where does this come from? Model documentation doesn't seem to define such an enumeration.

Reply:
Range, ok. But this isn't an enumeration.